### DISCUSSION TITLE
remaining fuel sensor in HA

### DISCUSSION THREAD

**Discussion by jackshaw20** [bimmerconnected/bimmer_connected#249] [Q&A] [answered]
I have recently changed my vehicle, I have noticed on this one in home assistant the 'remaining fuel' keeps randomly going to 0miles. my other vehicle (still BMW) this problem is not happeing with this vehicle, neither did it with my old vehicle.  only difference is my new vehicle has the new BMW Live Cockpit.   please see attached photo of home assistant;
![image](https://user-images.githubusercontent.com/61296690/99151117-e2f84680-2690-11eb-818a-8b880acadcff.png)

**Comment by minimalisticMe** [comment]
Looks like a Default value, I cannot imagine your car ever having 0mi remaining fuel.

**Comment by jackshaw20** [comment]
I have looked into this a bit more and the data coming from BMW is fine, its a problem with a template sensor I have configured to change from KM to MI,  but not sure why it keeps getting random 0 miles as this has worked fine previously and also works fine on the other vehicle;

see below sensors, the X3 template is the one giving random 0s.

![image](https://user-images.githubusercontent.com/61296690/99153298-7f294a00-269f-11eb-9912-21de38891e17.png)

**Comment by gerard33** [comment, marked as answer]
Strange indeed that it works correct for the X2 template and not for the X3 template. Nice cars btw 😎

I have checked your template and if the sensor is not available it will get the value 0. Does your sensor `sensor.x3_m40i_mileage` gets unavailable sometimes?

You can try to use this in value_template as that checks if the value > 0 and otherwise does nothing.

```yaml
value_template: >
  {% if states('sensor.x3_m40i_mileage')|float > 0 %}
    {{ (states('sensor.x3_m40i_mileage')|float/1.60934)|round(0) }}
  {% endif %}
```

And just to be sure; if you set HA to imperial it will use miles for this integration.

**Reply by jackshaw20** [reply]
Thanks :) 
I will change the template to your suggestion and monitor it.  I have just checked the mileage and yes this also gets the same unavailable status but the X2 doesn't.     I only noticed it was doing this as I have a tablet on the wall,  when the fuel level is below 25miles it pops up on the home screen for a quick visual that the car needs to be filled.

I might delete these templates for the X3 and re-create them with a new template name see if it makes a difference.  the X3 templates originally were for my old vehicle  x3_xdrive20d so I did a 'find and replace'  to x3_m40i in my config file, I can't see why this would make a difference though.

this is the X2 template:
![image](https://user-images.githubusercontent.com/61296690/99155837-88241680-26b3-11eb-929c-0c5691527e56.png)


this is the x3 template:
![image](https://user-images.githubusercontent.com/61296690/99155848-9eca6d80-26b3-11eb-8fc4-5ab78a5fa978.png)


There was a reason why I can't use imperial in home assistant, can't remember what it was for now which is why I made these templates.

**Reply by gerard33** [reply]
Maybe something strange in the templates indeed, although the ones you shared are looking good to me.

One question on your example. Are you defining the below twice? Because that's only needed once (assuming you are not using packages).
```yaml
- platform: template
  sensors:
```
Not sure if that has any influence though.

**Reply by jackshaw20** [reply]
I deleted the X3 templates, restarted HA and re entered them exactly as they was and its now working as it should..  not sure why it did it, very strange;

![image](https://user-images.githubusercontent.com/61296690/99194429-3a70e200-2777-11eb-90c1-9d3a3f5dd40b.png)
